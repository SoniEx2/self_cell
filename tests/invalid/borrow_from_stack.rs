use once_self_cell::sync::OnceSelfCell;

fn borrow_from_stack<'a>(_: &()) -> &'a String {
    let stack_string = String::from("hello stack");
    &stack_string
}

fn main() {
    let c: OnceSelfCell<(), &String> = OnceSelfCell::new(());
    let _ = c.get_or_init_dependent(borrow_from_stack);
}
