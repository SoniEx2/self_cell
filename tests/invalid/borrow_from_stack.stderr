error[E0515]: cannot return reference to local variable `stack_string`
 --> $DIR/borrow_from_stack.rs:5:5
  |
5 |     &stack_string
  |     ^^^^^^^^^^^^^ returns a reference to data owned by the current function
